src/main/java/org/folio/service/inventory/InventoryBindingManager.java

```diff
@@ -0,0 +1,98 @@
+
+  private final CirculationRequestsRetriever circulationRequestsRetriever;
+  private final RestClient restClient;
+
+  public InventoryBindingManager(RestClient restClient, CirculationRequestsRetriever circulationRequestsRetriever) {
+    this.restClient = restClient;
+    this.circulationRequestsRetriever = circulationRequestsRetriever;
+  }
+
+  public Future<List<Piece>> getPiecesWithActiveRequests(List<Piece> pieces, RequestContext requestContext) {
+    if (logger.isDebugEnabled()) {
+      logger.debug("Filtering pieces with active requests: {}", pieces.stream().map(Piece::getId).toList());
+    }
+    var futures = pieces.stream()
+      .map(piece -> getPieceWithRequests(piece, requestContext))
+      .collect(Collectors.toList());
+    return GenericCompositeFuture.all(futures)
+      .map(f -> f.<Piece>list().stream()
+        .filter(Objects::nonNull)
+        .toList());
+  }
+
+  private Future<Piece> getPieceWithRequests(Piece piece, RequestContext requestContext) {
+    return circulationRequestsRetriever.getNumberOfRequestsByItemId(piece.getItemId(), requestContext)
+      .map(requests -> requests > 0 ? piece : null);
+  }
+
+  public Future<Void> cancelPieceItemRequests(Piece piece, RequestContext requestContext) {
+    return handleBindingPieces(piece, requestContext, reqId -> cancelRequest(reqId, requestContext));
+  }
+
+  public Future<Void> transferPieceItemRequests(Piece piece, RequestContext requestContext) {
+    return handleBindingPieces(piece, requestContext, reqId -> transferRequest(reqId, requestContext));
+  }
+
+  private Future<Void> handleBindingPieces(Piece piece, RequestContext requestContext, Function<String, Future<Void>> bindHandler) {
+    return circulationRequestsRetriever.getRequestIdsByItemId(piece.getItemId(), requestContext)
+      .compose(reqIds -> {
+        var futures = reqIds.stream().map(bindHandler).toList();
+        return GenericCompositeFuture.all(futures);
+      })
+      .mapEmpty();
+  }
+
+  private Future<Void> cancelRequest(String reqId, RequestContext requestContext) {
+    logger.info(REQUEST_CANCEL_MESSAGE, reqId);
+    RequestEntry requestEntry = new RequestEntry(String.format(REQUEST_ENDPOINT, reqId));
+    return restClient.delete(requestEntry, requestContext);
+  }
+
+  private Future<Void> transferRequest(String reqId, RequestContext requestContext) {
+    return createNewItemForRequest(reqId, requestContext)
+      .compose(itemId -> {
+        logger.info(REQUEST_MOVE_MESSAGE, reqId, itemId);
+        JsonObject jsonObject =  JsonObject.of(REQUEST_JSON_KEY, itemId);
+        RequestEntry requestEntry = new RequestEntry(String.format(REQUEST_MOVE_ENDPOINT, reqId));
+        return restClient.postJsonObject(requestEntry, jsonObject, requestContext);
+      })
+      .mapEmpty();
+
+  }
+
+  private Future<String> createNewItemForRequest(String requestId, RequestContext requestContext) {
+    throw new RuntimeException("Not implemented");
```

Comment:
Item would be created earlier in the flow, @azizbekxm is working on it, so in transferRequest() we can add param itemId

Reply:
Modified it